### ISSUE TITLE
Using VertexShader (CanvasItem) and drawing textures by GDScript causes graphics to jump to top left corner in 3.2.4.

### ISSUE THREAD

**Issue by takaturre** [godotengine/godot#46897]
**Godot version:**
Godot 3.2.4 rc1-4. Was working fine in Godot 3.2.3 stable.

**OS/device including version:**
Windows 10, GLES3. (GPU: Nvidia GTX1080)

**Issue description:**
When using GDScript to draw texture portions on a CanvasItem *and* using a Vertex Shader, the graphical content jumps to the top left corner of the screen. See the screenshot.
![problem_screenshot](https://user-images.githubusercontent.com/16601926/110796845-1767c200-8281-11eb-812d-04eaaaab22b5.JPG)

**Steps to reproduce:**
Just follow the reproduction project. But basically:
- Create a Node2D.
- Put a GDScript on it that gets Texture as input and draws it again. (In my case into multiple smaller portions --> to get smoother vertex shading.)
- Create a Material and into it a Shader with just: `VERTEX = VERTEX;`.
```
shader_type canvas_item;
void vertex() {
   VERTEX = VERTEX; // Uncommenting this, makes it work fine - as the shader is essentially bypassed.
}
```

**Minimal reproduction project:**
[TextureDrawVertexShaderProblem.zip](https://github.com/godotengine/godot/files/6123364/TextureDrawVertexShaderProblem.zip)

**UPDATE:**
A new simpler reproduction project:
[TextureDrawVertexShaderProblem.zip](https://github.com/godotengine/godot/files/6124178/TextureDrawVertexShaderProblem.zip)

The problem seems to be calling `draw_set_transform()`. When uncommenting it from the GDScript the problem disappears --> So the problem seems to be that drawing origin gets set according to screen coordinates..?

Some screenshots:
![problem_screenshot](https://user-images.githubusercontent.com/16601926/110814253-f0fe5280-8291-11eb-924f-a719fd1b3ea4.JPG)
![image](https://user-images.githubusercontent.com/16601926/110815477-26f00680-8293-11eb-9e3f-5f45de75a509.png)

**Comment by Calinou** [comment]
I can confirm this on 3.2.4rc3, both with the GLES3 and GLES2 renderers. Disabling batching solves the issue (both in GLES3 and GLES2).

**Comment by lawnjelly** [comment]
I am working on this. Let me know if you manage to make the project more 'minimum' that is always helpful.

**Comment by takaturre** [comment]
Okay thanks! I managed to make the project more minimum - see above.

**Comment by lawnjelly** [comment]
Ah thanks! :+1: The minimum reproduction wasn't bad, but every bit simpler makes it quicker to fix.

In general it's better for users to spend 5 mins extra whittling down projects rather than the core devs (who then have to spend time to understand how the projects work). I know that makes us sound lazy (and I am!), but it makes best use of everyone's time overall and gets the most bugs fixed! :grin:

I've pinned it down now and made a PR. Although it needs lots of testing.

**Comment by takaturre** [comment]
Yeah, I know, that's what I'm sorry for :) ... For some reason, I just cleaned all the totally nonrelated stuff away without trying to find what the problem is actually about (as far as I can tell). Anyway, thanks a lot !

**Comment by akien-mga** [comment]
Fixed by #46901.
